File: codeforces/900/prime-substraction.py
# https://codeforces.com/problemset/problem/1238/A
# Big O:
# Time complexity: O(n^2)
# Space complexity: O(1)

# Problem


def is_prime_number(number):
    for i in range(2, number):
        if (number % i == 0):
            return False
    return True


def is_divided_by_prime_number(number):
    for i in range(2, number + 1):
        if (is_prime_number(i) & number % i == 0):
            return True
    return False


def can_be_substracted_with_primes(first_number, second_number):
    # first_number is guaranteed to be greater than second_number
    difference = first_number - second_number
    return "YES" if is_divided_by_prime_number(difference) else "NO"
    # alternative O(1)
    # return "YES" if difference > 1 else "NO"


# Read input
test_cases = int(input())

for i in range(0, test_cases):
    first_numer, second_numer = map(int, input().split())
    print(can_be_substracted_with_primes(first_numer, second_numer))

File: codeforces/900/prime-substraction.js
// https://codeforces.com/problemset/problem/1238/A
// Big O:
// Time complexity:
// Space complexity:

// Read input
var readline = require("readline");
var rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

let testCases;
let responses = [];
rl.on('line', (input) => {

    if (testCases === undefined) {
        testCases = parseInt(input);
        return;
    }

    if (testCases > 0) {
        const numbers = input.split(" ");
        responses.push(canBeSubstracted(numbers[0], numbers[1]));
        testCases--;
    }

    if (testCases <= 0) {
        responses.forEach((response) => console.log(response));
        rl.close();
    }
});

// Problem
function canBeSubstracted(x, y) {
    const difference = x - y;
    return isDividedByPrimeNumber(difference) ? "YES" : "NO";
}

function isDividedByPrimeNumber(number) {
    for (let i = 2; i <= number; i++) {
        const isPrime = isPrimeNumber(i);
        if (!isPrime) {
            continue;
        }
        if (number % i === 0) {
            return true
        }
    }
    return false;
}

function isPrimeNumber(number) {
    for (let i = 2; i < number; i++) {
        if (number % i === 0) {
            return false;
        }
    }

    return true;
}